### ISSUE TITLE
[DX] Figure out how to make a local backup

### ISSUE THREAD

**Issue by ykdojo** [ykdojo/defaang#240]
### Description

Now that we have more questions in our production database (ref: #232), we should figure out how to make a local backup.

Current plan: follow [this](https://ironeko.com/posts/creating-a-local-backup-of-a-supabase-database).

**Comment by aakashraj01** [comment]
Hey @ykdojo Can you assign me this issue under Hacktoberfest 2022 label? I would love to contribute in this issue

**Comment by ykdojo** [comment]
Update:

I tried the methods described in [the blog post](https://ironeko.com/posts/creating-a-local-backup-of-a-supabase-database), but it didn't quite work - I got a connection error.

I used pgAdmin and it seems like I was able to get a backup, but I had trouble restoring the production/dev databases from the backups, too.

**Comment by aakashraj01** [comment]
> Update:
> 
> I tried the methods described in [the blog post](https://ironeko.com/posts/creating-a-local-backup-of-a-supabase-database), but it didn't quite work - I got a connection error.
> 
> I used pgAdmin and it seems like I was able to get a backup, but I had trouble restoring the production/dev databases from the backups, too.

I also tried but error is coming always.

**Comment by CerealPlayer** [comment]
Hey @ykdojo I managed to make this work on my copy of defaang db in Supa bootstraped with supabase.sql file.

Instead of using the connection string (which errors because it needs a tls certificate, it could still be done), I used the pooling connection string, which is below the connection string in the settings panel. Pooling must be enabled.

![poolingconnstring](https://user-images.githubusercontent.com/41582680/194690337-34e9eca1-3656-4815-8249-c3d9eadfa8fa.png)

Worth noting I run pg_dump in a Postgres docker container, much easier to setup than installing Postgres locally.

**Comment by ykdojo** [comment]
@CerealPlayer sounds good. Were you able to restore a DB from that?

**Comment by CerealPlayer** [comment]
Yes! it generated a dump.sql with all the info I had stored in my copy of defaang.

**Comment by CerealPlayer** [comment]
[This](https://github.com/supabase/supabase/discussions/773#discussioncomment-401840) helps. The best approach would be to pg_dump certain tables that we know are critical and then pg_restore only those, since a full pg_dump would backup all schemas, and supabase doesn't support creating an empty database yet.

**Comment by ykdojo** [comment]
> The best approach would be to pg_dump certain tables that we know are critical and then pg_restore only those, since a full pg_dump would backup all schemas, and supabase doesn't support creating an empty database yet.

Got it. Backing up a database and restoring it seems like a pretty basic feature to have, though (I remember I had it in Heroku when I was using it).
